frontend/src/types/vuex.d.ts

```diff
@@ -0,0 +1,56 @@
+import { Store } from 'vuex';
+import { AuthorshipFile, User } from './types';
+
+interface AuthorshipInfo {
+  author: string;
+  checkedFileTypes?: string[];
+  files: AuthorshipFile[];
+  isMergeGroup: boolean;
+  isRefresh?: boolean;
+  location: string | undefined;
+  maxDate: string;
+  minDate: string;
+  name: string;
+  repo: string;
+}
+
+interface ZoomInfo {
+  isRefreshing?: boolean;
+  zAuthor: string;
+  zAvgCommitSize: number | string;
+  zFileTypeColors: { [key: string]: string };
+  zFilterGroup: string;
+  zFilterSearch: string;
+  zFromRamp: boolean;
+  zIsMerged: boolean;
+  zLocation: string | undefined;
+  zRepo: string;
+  zSince: string;
+  zTimeFrame: string;
+  zUntil: string;
+  zUser: User;
+}
+
+interface SummaryDates {
+  since: string;
+  until: string;
+}
+
+interface StoreState {
+  tabAuthorshipInfo: AuthorshipInfo;
+  tabZoomInfo: ZoomInfo;
+  summaryDates: SummaryDates;
+  mergedGroups: string[]
```

Comment:
Just a comment regarding format consistency. I think the semicolons are missing here. Actually, the current eslint rules do not seem to enforce the presence of semicolons, which can be discussed in a separate issue

Reply:
Thanks for pointing this out! Just fixed the semicolons in the latest commits.

Just checked the ESLint rules that we're adding in #1912 - the ruleset does enforce semicolons for regular statements (the [semi](https://typescript-eslint.io/rules/semi/) rule), but not for TypeScript `interface`s and `type`s specifically, as you mentioned. 

We can enforce this via the [member-delimiter-style](https://typescript-eslint.io/rules/member-delimiter-style) rule, i.e. just adding the following to `.eslintrc.json`.
```
"rules": {
  "@typescript-eslint/member-delimiter-style": "error"
}
```
Will open up a separate issue about this where we can investigate rules like the above one. Thanks!